Source/WebCore/html/CanvasBase.h

```diff
@@ -72,14 +73,14 @@ class CanvasDisplayBufferObserver : public CanMakeWeakPtr<CanvasDisplayBufferObs
     virtual void canvasDisplayBufferPrepared(CanvasBase&) = 0;
 };
 
-class CanvasBase {
+class CanvasBase : public AbstractRefCountedAndCanMakeWeakPtr<CanvasBase> {
 public:
     virtual ~CanvasBase();
 
     virtual void refCanvasBase() const = 0;
     virtual void derefCanvasBase() const = 0;
```

Comment:
It's possible we can also get rid of these and just use `ref`/`deref` as the virtual functions since we are adding virtual versions of them. Might be a nice cleanup and would be slightly better for performance.